utils/check_copies.py

```diff
@@ -166,6 +167,113 @@ def check_copies(overwrite: bool = False):
             + diff
             + "\nRun `make fix-copies` or `python utils/check_copies --fix_and_overwrite` to fix them."
         )
+    check_model_list_copy(overwrite=overwrite)
+
+
+def get_model_list():
+    """ Extracts the model list from the README. """
+    # If the introduction or the conclusion of the list change, the prompts may need to be updated.
+    _start_prompt = "🤗 Transformers currently provides the following architectures"
+    _end_prompt = "1. Want to contribute a new model?"
+    with open(os.path.join("README.md"), "r", encoding="utf-8") as f:
+        lines = f.readlines()
+    # Find the start of the list.
+    start_index = 0
+    while not lines[start_index].startswith(_start_prompt):
+        start_index += 1
+    start_index += 1
+
+    result = []
+    current_line = ""
+    end_index = start_index
+
+    while not lines[end_index].startswith(_end_prompt):
+        if lines[end_index].startswith("1."):
+            if len(current_line) > 1:
+                result.append(current_line)
+            current_line = lines[end_index]
+        elif len(lines[end_index]) > 1:
+            current_line = f"{current_line[:-1]} {lines[end_index].lstrip()}"
+        end_index += 1
+    if len(current_line) > 1:
+        result.append(current_line)
+
+    return "".join(result)
+
+
+def split_long_line_with_indent(line, max_per_line, indent):
+    """ Split the `line` so that it doesn't go over `max_per_line` and adds `indent` to new lines. """
+    words = line.split(" ")
+    lines = []
+    current_line = words[0]
+    for word in words[1:]:
+        if len(f"{current_line} {word}") > max_per_line:
+            lines.append(current_line)
+            current_line = " " * indent + word
+        else:
+            current_line = f"{current_line} {word}"
+    lines.append(current_line)
+    return "\n".join(lines)
+
+
+def convert_to_rst(model_list, max_per_line=None):
+    """ Convert `model_list` to rst format. """
+    # Convert **[description](link)** to `description <link>`__
+    model_list = re.sub(r"\*\*\[([^\]]*)\]\(([^\)]*)\)\*\*", r"`\1 <\2>`__", model_list)
```

Comment:
That feeling when this regex works on the first try ;-)

Reply:
Very nice!